Feature/generate salts command

Jaspervv opened the pull request:
These two commands come straight from our Themosis Plugin.

### Why
see this discussion themosis/themosis#265


### What's included
adds two console commands:

**env:set** sets something in your `.env` file.  
It can be used as `env:set key=value` or `env:set key value`.

**salts:generate** uses `env:set` to generate WordPress salts.


### Source
The included **env:set** command is an edited version of the file [here](https://github.com/imliam/laravel-env-set-command/blob/be51ce036f8d5c935a5bb2d21768831a4efd0d17/src/EnvironmentSetCommand.php).  
I made edits for support with spaces, plus some extra checks and what not.


### Improvements
Code might not be up to standard, but it works. Function could use some improvements like using `Illuminate\Filesystem\Filesystem` methods instead of core PHP functions.

jlambe commented:
What about to support the new sodium library coming on PHP7.2 for the salts ? :
http://php.net/manual/fr/sodium.installation.php

Jaspervv commented:
Thanks for the suggestion, I modified the `salt()` function in edf60d6a13669ecd571803579b441b3098d3d3f5.  
Im not familiar with Sodium so I hope I implemented it correctly. :?

jlambe commented:
@Jaspervv Sounds like WordPress is coming with a compat sodium library for backward compatibility: https://core.trac.wordpress.org/ticket/45806#comment:8 Not sure yet how this could impact the code for this feature in creating secure salt keys.

Jaspervv commented:
Is this going to be a separate library or a set of new functions in the WordPress core? If it's a set of new functions it won't affect this feature because Console Commands can't make use of WordPress functions.

jlambe commented:
@Jaspervv It should be include in WordPress core so yes it won't affect the command but as the library needs PHP 7.2 at the minimum, I was intrigued by the sodium_compat library that could also be used in order to provide the feature on PHP 7.1. -> just for information ;)

Jaspervv commented:
in the link you sent one dude says WordFence also uses sodium_compat, I looked at their source and they seem to use this library:
https://github.com/paragonie/sodium_compat

jlambe commented:
@Jaspervv I'm going to implement your PR but differently as there is already a third-party popular package to handle environment variables. I think that if people would like to be able to set environment variables through the command-line, they can rely on the following package:
- https://github.com/imliam/laravel-env-set-command

So I'm going to close this PR but part of your code is still going to come on next release. Thanks for your work.
